### PR TITLE
update to typescript@2.1.0-dev.20160827

### PR DESCRIPTION
**Pull request by IgorMinar**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,14 +26,14 @@
...
-    "gulp-tslint": "^4.3.4",
+    "gulp-tslint": "^6.1.1",
     "gulp-typescript": "^2.7.6",
     "gulp-util": "^3.0.4",
     "merge2": "^0.3.1",
     "temp": "^0.8.1",
     "tsd": "^0.6.0",
-    "tslint": "^3.7.1",
-    "typescript": "2.0.0"
+    "tslint": "^3.15.0-dev.0",
```
Comment (IgorMinar) [inline package.json]: because of peerDeps I had to bump tslint as well

File: src/decorator-annotator.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ class ClassRewriter extends Rewriter {
   }
 
   private decoratorsToLower(n: ts.Node): ts.Decorator[] {
-    return (n.decorators || []).filter(d => this.shouldLower(d));
+    return (n.decorators as ts.NodeArray<ts.Decorator>|| []).filter(d => this.shouldLower(d));
```
Comment (IgorMinar) [inline src/decorator-annotator.ts]: I'm not quire sure why this change is required

File: src/tsickle.ts [hunk 3]
```diff
@@ -749,18 +749,7 @@ class ExternsWriter extends ClosureRewriter {
...
-                name = getIdentifierText(decl.name as ts.Identifier);
-                break;
-              case ts.SyntaxKind.StringLiteral:
-                name = (decl.name as ts.StringLiteral).text;
-                break;
-              default:
-                this.errorUnimplementedKind(decl.name, 'namespace name');
-                break;
-            }
+            let name = getIdentifierText(decl.name as ts.Identifier);
```
Comment (IgorMinar) [inline src/tsickle.ts]: decl.name.kind is already of type ts.SyntaxKind.Identifier so this whole switch block is not needed.

File: src/type-translator.ts [hunk 4]
```diff
@@ -27,17 +27,27 @@ function isClosureProvidedType(symbol: ts.Symbol): boolean {
...
-    ts.TypeFlags.Any,           ts.TypeFlags.String,
-    ts.TypeFlags.Number,        ts.TypeFlags.Boolean,
-    ts.TypeFlags.Void,          ts.TypeFlags.Undefined,
-    ts.TypeFlags.Null,          ts.TypeFlags.Enum,
-    ts.TypeFlags.StringLiteral, ts.TypeFlags.TypeParameter,
-    ts.TypeFlags.Class,         ts.TypeFlags.Interface,
-    ts.TypeFlags.Reference,     ts.TypeFlags.Tuple,
-    ts.TypeFlags.Union,         ts.TypeFlags.Intersection,
-    ts.TypeFlags.Anonymous,     ts.TypeFlags.Instantiated,
-    ts.TypeFlags.ObjectLiteral, ts.TypeFlags.ESSymbol,
```
Comment (IgorMinar) [inline src/type-translator.ts]: I just removed `ObjectLiteral` because that's not marked as `@internal` - I don't know what kind of implications this has on the rest of the code - I couldn't find a good replacement for this type in typescript.d.ts. please advice if this is the right change.

File: tsconfig.json [hunk 5]
```diff
@@ -10,7 +10,6 @@
     "strictNullChecks": true
   },
   "files": [
-    "node_modules/typescript/lib/lib.es2015.collection.d.ts",
```
Comment (IgorMinar) [inline tsconfig.json]: this is not needed since we use `lib` above

File: src/decorator-annotator.ts (hunk 2 above)
Comment (mprobst) [inline src/decorator-annotator.ts]: I think `node.decorators` ends up as some odd union type because it's defined multiple times in the dependency hierarchy of `ts.Node`. No idea why it happens in this version though.

File: src/tsickle.ts [hunk 6]
```diff
@@ -749,18 +749,7 @@ class ExternsWriter extends ClosureRewriter {
...
             if (name === undefined) break;
```
Comment (mprobst) [inline src/tsickle.ts]: can name still be `undefined` here, then?

File: src/type-translator.ts (hunk 4 above)
Comment (mprobst) [inline src/type-translator.ts]: That's _now_ marked as `@internal`? No idea, maybe @chuckjaz knows?

**Comment by rkirov** [comment]
closing in favor of https://github.com/angular/tsickle/pull/196

File: src/type-translator.ts (hunk 4 above)
Comment (evmar) [inline src/type-translator.ts]: This function is only used for debugging anyway so it's likely fine.
